### ISSUE TITLE
Bug: inconsistency for the BatchNormalization layer between Tensorflow, CNTK and Theano backends #12957

### ISSUE THREAD

**Issue by gowthamkpr** [rthadur/recaptcha#313]
Library | Version
-- | --
python | 3.6.8
GCC | 7.3.0
tensorflow-base/tensorflow-gpu | 1.13.1
keras-gpu/keras-base | 2.2.4
theano | 1.0.4
cudnn | 7.3.1
cudatoolkit | 10.0.130

Machine Configuration：

Ubuntu 18.04.2 LTS

Intel(R) Xeon(R) CPU E5-2640 v4 @ 2.40GHz 128GB RAM

CUDA Version: 10.2

Double GeForce GTX 1080ti

Description
This bug happened when I was using a pre-trained keras model called Xception. I loaded the same model on different backends(tensorflow, CNTK and theano). When I tried to do some prediction, I found that the prediction accuracy of tensorflow are different from that of CNTK and theano. Among 1500 images, the number of images for which tensorflow and CNTK have different results is 118, the number of images for which tensorflow and theano have different results is also 118, which theano and CNTK do not have any different result.

Some of the prediction results of this model on 3 backends are shown as below.

It seems that there is something wrong with tensorflow. I put images into the model and recorded the ouputs of each layers to see how this happens. It seems that the outputs on CNTK and theano are very close, but for tensorflow, the outputs of a BatchNormalization layer and all the layers after this layer are very different from CNTK and theano.

The results are attached below.

The delta calumn shows the discrepancy, which is calculated as:
image

Based on the results, the deviation seems to be caused by batch_normalization_2 layer of tensorflow backend.
